src/app/tests/TestReadInteraction.cpp

```diff
@@ -2228,8 +2264,12 @@ void TestReadInteraction::TestSubscribeInvalidAttributePathRoundtrip(nlTestSuite
         NL_TEST_ASSERT(apSuite, engine->ActiveHandlerAt(0) != nullptr);
         delegate.mpReadHandler = engine->ActiveHandlerAt(0);
 
-        delegate.mpReadHandler->SetStateFlag(ReadHandler::ReadHandlerFlags::WaitingUntilMinInterval, false);
-        delegate.mpReadHandler->SetStateFlag(ReadHandler::ReadHandlerFlags::WaitingUntilMaxInterval, false);
+        ReadHandlerNode * delegateNode = reportScheduler->FindReadHandlerNode(delegate.mpReadHandler);
+        delegateNode->SetTestFlags(ReadHandlerNode::TestFlags::MinIntervalElapsed, true);
+        delegateNode->SetTestFlags(ReadHandlerNode::TestFlags::MaxIntervalElapsed, true);
+        // Manually trigger the callback that would schedule the next report as it would normally have been called if the time had
+        // elapsed as simulated above
+        delegateNode->RunCallback();
```

Comment:
This sort of thing makes it really easy to cover up actual impl bugs.

We should very seriously think about _actually testing the code we ship_ here.  I know this test already had a bunch of problems in that area....

Reply:
This should be addressed in : https://github.com/project-chip/connectedhomeip/issues/28129

The goal here was to try to change the behavior of the test as little as possible for the integration.